clustered/client/src/main/java/org/ehcache/clustered/client/config/builders/ClusteringServiceConfigurationBuilder.java

```diff
@@ -58,11 +67,110 @@ public ServerSideConfigurationBuilder autoCreate() {
+   *
+   * @throws NullPointerException if {@code unit} is {@code null}
+   * @throws IllegalArgumentException if {@code amount} is negative
+   */
+  public ClusteringServiceConfigurationBuilder readOperationTimeout(long duration, TimeUnit unit) {
+    return new ClusteringServiceConfigurationBuilder(this, TimeoutDuration.of(duration, unit));
   }
 
   @Override
   public ClusteringServiceConfiguration build() {
-    return new ClusteringServiceConfiguration(clusterUri);
+    return new ClusteringServiceConfiguration(clusterUri, readOperationTimeout);
+  }
+
+  /**
+   * Internal method to build a new {@link ClusteringServiceConfiguration} from the {@link ServerSideConfigurationBuilder}.
+   *
+   * @param clientSideConfiguration the {@code ClusteringServiceClientSideConfiguration} passed to the
+   *                                {@link ServerSideConfigurationBuilder#ServerSideConfigurationBuilder(ClusteringServiceClientSideConfiguration)
+   *                                ServerSideConfigurationBuilder(ClusteringServiceClientSideConfiguration)}
+   *                                constructor
+   * @param serverSideConfiguration the {@code ServerSideConfiguration} to use
+   *
+   * @return a new {@code ClusteringServiceConfiguration} instance built from {@code clientSideConfiguration} and
+   *        {@code serverSideConfiguration}
+   */
+  static ClusteringServiceConfiguration build(ClusteringServiceClientSideConfiguration clientSideConfiguration,
+                                              ServerSideConfiguration serverSideConfiguration) {
+    return new ClusteringServiceConfiguration(
+        clientSideConfiguration.getClusterUri(),
+        clientSideConfiguration.getReadOperationTimeout(),
+        clientSideConfiguration.isAutoCreate(),
+        serverSideConfiguration);
+  }
+
+  /**
+   * The client-side portion of the {@link ClusteringServiceConfiguration} used during
+   * configuration building.
+   */
+  private static class ClusteringServiceClientSideConfigurationImpl implements ClusteringServiceClientSideConfiguration {
+    private final URI clusterUri;
+    private final boolean autoCreate;
+    private final TimeoutDuration readOperationTimeout;
+
+    ClusteringServiceClientSideConfigurationImpl(URI clusterUri, boolean autoCreate, TimeoutDuration readOperationTimeout) {
+      this.clusterUri = clusterUri;
+      this.autoCreate = autoCreate;
+      this.readOperationTimeout = readOperationTimeout;
+    }
+
+    @Override
+    public URI getClusterUri() {
+      return clusterUri;
+    }
+
+    @Override
+    public boolean isAutoCreate() {
+      return autoCreate;
+    }
+
+    @Override
+    public TimeoutDuration getReadOperationTimeout() {
+      return readOperationTimeout;
+    }
+  }
+
+  /**
+   * The client-side portion of the {@link ClusteringServiceConfiguration} used during
+   * configuration building.
+   */
+  interface ClusteringServiceClientSideConfiguration {
```

Comment:
We could go back to `ServerSideConfigurationBuilder` having to be aware of all of the properties in  `ClusteringServiceConfigurationBuilder`.  We could pass around the `ClusteringServiceConfigurationBuilder` instance instead ... thoughts?